nylas/models/attachments.py

```diff
@@ -0,0 +1,47 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from dataclasses_json import dataclass_json
+from typing_extensions import TypedDict, NotRequired
+
+
+@dataclass_json
+@dataclass
+class Attachment:
+    """
+    An attachment on a message.
+
+    Attributes:
+        id: Globally unique object identifier.
+        size: Size of the attachment in bytes.
+        filename: Name of the attachment.
+        content_type: MIME type of the attachment.
+    """
+
+    id: str
+    size: int
+    filename: Optional[str] = None
+    content_type: Optional[str] = None
+
+
+class CreateAttachmentRequest(TypedDict):
+    """
+    A request to create an attachment.
+
+    Attributes:
+        filename: Name of the attachment.
+        content_type: MIME type of the attachment.
+        content: Base64 encoded content of the attachment.
+        size: Size of the attachment in bytes.
+        content_id: The content ID of the attachment.
+        content_disposition: The content disposition of the attachment.
+        is_inline: Whether the attachment is inline.
+    """
+
+    filename: str
+    content_type: str
+    content: str
+    size: int
+    content_id: NotRequired[str]
+    content_disposition: NotRequired[str]
```

Comment:
I'm curious—is `content_disposition` part of the schema for attachment objects, or just part of these creation requests? Have you tested what fields the API sends back when fetching a message with an attachment? (I didn't)

Reply:
Yup, it exists on creation and retrieval. Gere's an example response from the `GET /messages` endpoint with an attachment:

```json
{
    "id": "snip",
    "grant_id": "snip",
    "filename": "invite.ics",
    "size": 2420,
    "content_type": "application/ics; name=\"invite.ics\"",
    "is_inline": false,
    "content_disposition": "attachment; filename=\"invite.ics\""
}
```

Note that `is_inline` is also present.